packages/nextjs/pages/subscribe/[planId].tsx

```diff
@@ -0,0 +1,73 @@
+import { useEffect, useState } from "react";
+import { useRouter } from "next/router";
+import toast from "react-hot-toast";
+import { Button } from "~~/components/Button";
+import { Loader } from "~~/components/Loader";
+import PlanDetailsBox from "~~/components/PlanDetailsBox";
+import { Plan } from "~~/context/Types";
+import { useWeb3AuthContext } from "~~/context/Web3AuthContext";
+
+const SubscriptionDetails = () => {
+  const router = useRouter();
+  const [isLoading, setIsLoading] = useState(true);
+  const [plan, setPlan] = useState<Plan>();
+  const { getPlans, isConnected, purchasePlan, subsContract } = useWeb3AuthContext();
+
+  const planId = router.query.planId as string;
+
+  const redirectToSubscribeLanding = () => {
+    router.push("/subscribe");
+  };
+
+  const handleOnPurchaseAttempt = (plan: Plan) => {
+    if (plan.planId) {
+      if (!isConnected) {
+        toast.error("Please log in before submitting a purchase");
+        return;
+      }
+      const toastId = toast.loading("Wait some moments to complete the purchase!");
+      // @note: should ask before confirmation
+      purchasePlan(plan.planId, Number(plan.price), plan.paymentTokenAddress);
+      toast.dismiss(toastId);
+    }
+  };
+
+  useEffect(() => {
+    const setPlanIfExists = async () => {
+      const plans = await getPlans();
+      const foundPlan = plans?.find(plan => plan.planId === planId);
+
+      if (foundPlan) {
+        setPlan(foundPlan);
+      }
+      setIsLoading(false);
+    };
+
+    if (subsContract && !plan && planId) {
+      setPlanIfExists();
+    }
+  }, [plan, planId, getPlans, subsContract]);
+
+  return plan ? (
+    <div className="p-32 flex-grow" data-theme="exampleUi">
+      <h1 className="text-4xl sm:text-6xl">Subscribe to planId: {planId}</h1>
```

Comment:
Maybe rather than putting the plan id in the title I would put the website.

Reply:
I think that complicates things, since some people may put it with `https://` or `http` or without `www` etc. So we should support multiple interpretations and that can be tricky. **I'd suggest to stick with `planId`'s**

If we go that direction you suggested, some changes need to be made to this feature since it doesn't support `https://www.` etc because it understands that the `/` are subfolders (I think there's a solution for that but haven't tested it yet)

Reply:
Yes that makes a lot of sense, let me change that. Thanks!